src/nbl/video/ILogicalDevice.cpp

```diff
@@ -29,6 +29,62 @@ core::smart_refctd_ptr<IGPUDescriptorSetLayout> ILogicalDevice::createDescriptor
     return createDescriptorSetLayout_impl(_begin,_end);
 }
 
+bool ILogicalDevice::updateDescriptorSets(uint32_t descriptorWriteCount, const IGPUDescriptorSet::SWriteDescriptorSet* pDescriptorWrites, uint32_t descriptorCopyCount, const IGPUDescriptorSet::SCopyDescriptorSet* pDescriptorCopies)
+{
+    for (auto i = 0; i < descriptorWriteCount; ++i)
+    {
+        auto* ds = static_cast<IGPUDescriptorSet*>(pDescriptorWrites[i].dstSet);
+        ds->incrementVersion();
+
+        auto* descriptors = ds->getDescriptors(pDescriptorWrites[i].descriptorType, pDescriptorWrites[i].binding);
+        auto* samplers = ds->getMutableSamplers(pDescriptorWrites[i].binding);
+        for (auto j = 0; j < pDescriptorWrites[i].count; ++j)
+        {
+            descriptors[j] = pDescriptorWrites[i].info[j].desc;
+
+            if (samplers)
+                samplers[j] = pDescriptorWrites[i].info[j].info.image.sampler;
+        }
+    }
+
+    for (auto i = 0; i < descriptorCopyCount; ++i)
+    {
+        const auto* srcDS = static_cast<const IGPUDescriptorSet*>(pDescriptorCopies[i].srcSet);
+        auto* dstDS = static_cast<IGPUDescriptorSet*>(pDescriptorCopies[i].dstSet);
+
+        auto foundBindingInfo = std::lower_bound(srcDS->getLayout()->getBindings().begin(), srcDS->getLayout()->getBindings().end(), pDescriptorCopies[i].srcBinding,
```

Comment:
same as https://github.com/Devsh-Graphics-Programming/Nabla/pull/345/files#r1047776314 probably wouldn't even need to use many getters

Reply:
Related https://github.com/Devsh-Graphics-Programming/Nabla/pull/345#discussion_r1056113772